internal/types/client.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/ory/fosite"
+	v1 "go.infratographer.com/identity-api/pkg/api/v1"
+)
+
+// OAuthClient is an OAuth 2.0 Client
+type OAuthClient struct {
+	ID       string
+	TenantID string
+	Name     string
+	Secret   string
+	Audience []string
+	Scope    string
```

Comment:
We haven't really brought scope into the picture here yet and I'm not sure we plan to for identity-api. Might be worth adding this later once we've made a decision there.

Reply:
Sounds good.